frontend/src/tests/import-export.test.js

```diff
@@ -107,6 +201,146 @@ describe("Import/export library functionality", () => {
+
+        // export instructor data to a JSON File object
+        const fileJSON = getInstructorsDataFile("json");
+        expect(fileJSON.name).toEqual(
+            `instructors_export_${new Date().toLocaleDateString("en-CA", {
+                year: "numeric",
+                month: "numeric",
+                day: "numeric",
+            })}.json`
+        );
+
+        // export instructor data to a XLSX File object
+        const fileXLSX = getInstructorsDataFile("spreadsheet");
+        expect(fileXLSX.name).toEqual(
+            `instructors_export_${new Date().toLocaleDateString("en-CA", {
+                year: "numeric",
+                month: "numeric",
+                day: "numeric",
+            })}.xlsx`
+        );
+
+        // invalid file type, should throw error
+        expect(() => getInstructorsDataFile("nonsense")).toThrow(Error);
+
+        const resultCorrect = instructorData.map((instructor) =>
+            prepareMinimal.instructor(instructor)
+        );
+
+        // import instructor data from a JSON File object
+        const dataJSON = JSON.parse(fileJSON.buffer.toString()).instructors;
+        const resultJSON = normalizeImport(
+            {
+                fileType: "json",
+                data: dataJSON,
+            },
+            instructorSchema
+        );
+        expect(resultJSON).toEqual(resultCorrect);
+
+        // import instructor data from a XLSX File object
+        const workbook = XLSX.read(fileXLSX.buffer[0], { type: "array" });
+        const sheet = workbook.Sheets[workbook.SheetNames[0]];
+        let dataXLSX = XLSX.utils.sheet_to_json(sheet, { header: 1 });
+        // transform to array of objects
+        const keys = dataXLSX.shift();
+        dataXLSX = dataXLSX.map(function (row) {
+            return keys.reduce(function (obj, key, i) {
+                obj[key] = row[i];
+                return obj;
+            }, {});
+        });
+        const resultXLSX = normalizeImport(
+            {
+                fileType: "spreadsheet",
+                data: dataXLSX,
+            },
+            instructorSchema
+        );
+        expect(resultXLSX).toEqual(resultCorrect);
+
+        // import instructor data from a CSV File object
+        const workbook1 = XLSX.read(fileCSV.buffer[0], { type: "array" });
+        const sheet1 = workbook1.Sheets[workbook1.SheetNames[0]];
+        let dataCSV = XLSX.utils.sheet_to_json(sheet1, { header: 1 });
+        // transform to array of objects
+        const keys1 = dataCSV.shift();
+        dataCSV = dataCSV.map(function (row) {
+            return keys1.reduce(function (obj, key, i) {
+                obj[key] = row[i];
+                return obj;
+            }, {});
+        });
+        const resultCSV = normalizeImport(
+            {
+                fileType: "spreadsheet",
+                data: dataCSV,
+            },
+            instructorSchema
+        );
+        expect(resultCSV).toEqual(resultCorrect);
```

Comment:
This tests for a round trip, but it doesn't test that the output has the required fields. We need to test the CSV to make sure it has first names/last names/etc. for the required data. This will be happening in every test, so you might make a utility function that's something like `csvContainsData(csv, requiredData)` to test if a CSV has what it's supposed to.

A big reason for this test is to catch bugs that we had before where, for example, the export function exported `fist_name` twice and never exported `last_name`

Reply:
Instead of going round trip (using normalizeImport to get the data back and check), should I manually check that the "File" object created contains all columns from the data and each cell is identical with the data?